Source/ACE/Entity/Cow.cs

```diff
@@ -0,0 +1,126 @@
+﻿using ACE.Entity.Enum;
+using ACE.Entity.Actions;
+using ACE.Network.Enum;
+using ACE.Network.GameEvent.Events;
+using ACE.Network.Motion;
+using ACE.Entity.Enum.Properties;
+using System;
+using ACE.DatLoader.FileTypes;
+using ACE.Network.GameMessages.Messages;
+using ACE.Common;
+
+namespace ACE.Entity
+{
+    public class Cow : Creature
+    {
+        private static readonly UniversalMotion motionTipRight = new UniversalMotion(MotionStance.Standing, new MotionItem(MotionCommand.TippedRight));
+
+        public Cow(AceObject aceO)
+            : base(aceO)
+        {
+            var weenie = Database.DatabaseManager.World.GetAceObjectByWeenie(AceObject.WeenieClassId);
+
+            UseRadius = 1;
+            IsAlive = true;
+            SetupVitals();
+        }
+
+        private double? resetTimestamp;
+        private double? ResetTimestamp
+        {
+            get { return resetTimestamp; }
+            set { resetTimestamp = Time.GetTimestamp(); }
+        }
+
+        private double? useTimestamp;
+        private double? UseTimestamp
+        {
+            get { return useTimestamp; }
+            set { useTimestamp = Time.GetTimestamp(); }
+        }
+
+        private uint? AllowedActivator
+        {
+            get;
+            set;
+        }
+
+        public override void HandleActionOnUse(ObjectGuid playerId)
+        {
+            ActionChain chain = new ActionChain();
+            CurrentLandblock.ChainOnObject(chain, playerId, (WorldObject wo) =>
+            {
+                Player player = wo as Player;
+                if (player == null)
+                {
+                    return;
+                }
+
+                SetupModel csetup = SetupModel.ReadFromDat(SetupTableId.Value);
+                float radiusSquared = (UseRadius.Value + csetup.Radius) * (UseRadius.Value + csetup.Radius);
+                float playerDistanceTo = player.Location.SquaredDistanceTo(Location);
+
+                ////if (playerDistanceTo >= 2500)
+                ////{
+                ////    var sendTooFarMsg = new GameEventDisplayStatusMessage(player.Session, StatusMessageType1.Enum_0037);
+                ////    player.Session.Network.EnqueueSend(sendTooFarMsg, sendUseDoneEvent);
+                ////    return;
+                ////}
+
+                if (playerDistanceTo >= radiusSquared)
```

Comment:
a reasonable suggestion, it probably will be in at least a few more objects as well